dhctl/pkg/preflight/preflight.go

```diff
@@ -16,17 +16,24 @@ package preflight
 
 import (
 	"github.com/deckhouse/deckhouse/dhctl/pkg/app"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/kubernetes/actions/deckhouse"
 	"github.com/deckhouse/deckhouse/dhctl/pkg/log"
 	"github.com/deckhouse/deckhouse/dhctl/pkg/system/ssh"
 )
 
 type PreflightCheck struct {
-	sshClient *ssh.Client
+	sshClient               *ssh.Client
+	installConfig           *deckhouse.Config
+	imageDescriptorProvider imageDescriptorProvider
+	buildDigestProvider     buildDigestProvider
 }
 
-func NewPreflightCheck(sshClient *ssh.Client) PreflightCheck {
+func NewPreflightCheck(sshClient *ssh.Client, config *deckhouse.Config) PreflightCheck {
 	return PreflightCheck{
-		sshClient: sshClient,
+		sshClient:               sshClient,
+		installConfig:           config,
+		imageDescriptorProvider: remoteDescriptorProvider{},
+		buildDigestProvider:     &dhctlBuildDigestProvider{DigestFilePath: app.DeckhouseImageDigestFile},
```

Comment:
Maybe it's worth creating constructor functions that will return the corresponding interfaces as a type?